app/src/main/java/io/github/androidacademyglobal/AcademyApplication.kt

```diff
@@ -1,18 +1,39 @@
 package io.github.androidacademyglobal
 
 import android.app.Application
+import android.util.Log
+import androidx.work.BackoffPolicy
+import androidx.work.Configuration
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.OneTimeWorkRequest
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
 import com.google.firebase.FirebaseApp
 import com.google.firebase.crashlytics.ktx.crashlytics
 import com.google.firebase.ktx.Firebase
+import io.github.aag.core.data.background.UpdateAuthTokenWorker
+import io.github.aag.core.data.background.UpdateAuthTokenWorkerFactory
 import io.github.aag.core.di.initKoin
+import io.github.aag.core.domain.repositories.AuthRepository
+import io.github.aag.core.domain.repositories.ProfileRepository
 import io.github.aakira.napier.DebugAntilog
 import io.github.aakira.napier.Napier
 import io.github.androidacademyglobal.di.appModule
 import io.github.androidacademyglobal.logging.CrashlyticsAntilog
+import org.koin.android.ext.android.inject
 import org.koin.android.ext.koin.androidContext
 import org.koin.android.ext.koin.androidLogger
+import java.util.concurrent.TimeUnit
+
+class AcademyApplication : Application(), Configuration.Provider {
+    private val updateAuthTokenRequest =
+        PeriodicWorkRequestBuilder<UpdateAuthTokenWorker>(2, TimeUnit.HOURS)
+            .setBackoffCriteria(
+                BackoffPolicy.LINEAR,
+                OneTimeWorkRequest.MIN_BACKOFF_MILLIS,
```

Comment:
Is this a good idea - retrying to get auth token every 10 seconds until user logs in/registers and we get `refreshToken` (to use it for getting an auth token)? 🤔          
I was also thinking about introducing some constraints - maybe it's better to run this background work only when user actually uses the app (to save some battery - i.e. we don't need to update auth token every 2 hours if user uses the app once per day)

Reply:
You don't need WorkManager for this case

Reply:
I need to have fresh `authToken` prior making any request to back-end. `authToken` expires every 2 hours. I thought that `workManager` is a good option for this case.